Shared/Supporting Files/Views/SearchResultView.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SearchResultView: View {
+    /// All samples retrieved from the Samples directory.
+    let samples: [Sample]
+    
+    /// The search query in the search bar.
+    let query: String
+    
+    /// The search result to display in the various sections.
+    @State private var searchResult = SearchResult(nameMatches: [], descriptionMatches: [], tagMatches: [])
```

Comment:
This doesn't need to be a state property because it is never updated by the view itself. Instead, the view should be initialized with this:

```swift
let searchResult: SearchResult
```

You can either create that outside of this view or the way it is now with an initializer:

```swift
extension SearchResultView {
    init(samples: [Sample], query: String) {
        let nameMatches: [Sample]
        let descriptionMatches: [Sample]
        let tagMatches: [Sample]
        
        if query.isEmpty {
            // Show all samples in the name section when query is empty.
            nameMatches = samples
            descriptionMatches = []
            tagMatches = []
        } else {
            // The names of the samples already found in a previous section.
            var previousSearchResults: Set<String> = []
            
            // Partially match a query to a sample's name.
            nameMatches = samples.filter { $0.name.localizedCaseInsensitiveContains(query) }
            previousSearchResults.formUnion(nameMatches.map(\.name))
            
            // Partially match a query to a sample's description.
            descriptionMatches = samples.filter { $0.description.localizedCaseInsensitiveContains(query) }
                .filter { !previousSearchResults.contains($0.name) }
            previousSearchResults.formUnion(descriptionMatches.map(\.name))
            
            // Match a query to one of the sample's tags.
            tagMatches = samples.filter { sample in
                sample.tags.contains { tag in
                    tag.localizedCaseInsensitiveCompare(query) == .orderedSame
                }
            }
            .filter { !previousSearchResults.contains($0.name) }
        }
        
        let searchResult = SearchResult(
            nameMatches: nameMatches,
            descriptionMatches: descriptionMatches,
            tagMatches: tagMatches
        )
        self.init(searchResult: searchResult)
    }
}
```

Then the `onAppear` and `onChange` modifiers aren't needed.